test/langtools/tools/lib/toolbox/ToolBox.java

```diff
@@ -185,6 +185,28 @@ public void checkEqual(List<String> l1, List<String> l2) throws Error {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Filters a list of strings which don't match the given regular expression.
+     * @param regex the regular expression
+     * @param lines the strings to be filtered
+     * @return the strings not matching the regular expression
+     */
+    public List<String> grepNotMatch(String regex, List<String> lines) {
+        return grepNotMatch(Pattern.compile(regex), lines);
+    }
+
+    /**
+     * Filters a list of strings which don't match the given regular expression.
+     * @param pattern the regular expression
+     * @param lines the strings to be filtered
+     * @return the strings not matching the regular expression
+     */
+    public List<String> grepNotMatch(Pattern pattern, List<String> lines) {
```

Comment:
Instead of new methods named `grepNotMatch` I suggest adding new overloads of `grep` that take an additional `boolean invert` parameter that is conceptually equivalent to the `grep` `-v` option.  The existing `grep` methods can be updated to delegate to the new methods, passing `false` for the new parameter.

Reply:
@jonathan-gibbons Thank you for your suggestion. The code is updated now.